modules/javafx.controls/src/test/java/test/javafx/scene/control/TreeTableViewTest.java

```diff
@@ -374,83 +374,133 @@ private String debug() {
+        TreeTableColumn<String, String> col = setupForPermutationTest();
         treeTableView.getSortOrder().add(col);
-        VirtualFlowTestUtils.assertListContainsItemsInOrder(treeTableView.getRoot().getChildren(), apple, banana, orange);
         verifySelectionAfterPermutation();
     }
 
     @Test public void testSelectionUpdatesCorrectlyAfterRootSetAll() {
-        initSortTestStructure();
         setupForPermutationTest();
-        treeTableView.getRoot().getChildren().setAll(banana, apple, orange);
+        reverseChildrenOrder(treeTableView.getRoot());
         verifySelectionAfterPermutation();
     }
 
     @Test public void testSelectionUpdatesCorrectlyAfterChildSetAll() {
-        initSortTestStructure();
         setupForPermutationTest();
-        banana.getChildren().setAll(banana.getChildren().get(2), banana.getChildren().get(0), banana.getChildren().get(1));
+        reverseChildrenOrder(((TreeItem<String>)sm.getSelectedItem()).getParent());
         verifySelectionAfterPermutation();
     }
 
-    private void setupForPermutationTest() {
+    private void reverseChildrenOrder(TreeItem<String> treeItem) {
+        List<TreeItem<String>> childrenReversed = new ArrayList<>();
+        int childrenSize = treeItem.getChildren().size();
+        for (int i = 0; i < childrenSize; i++) {
+            childrenReversed.add(treeItem.getChildren().get(childrenSize - 1 - i));
+        }
+        treeItem.getChildren().setAll(childrenReversed);
+    }
+
+    private TreeTableColumn<String, String> setupForPermutationTest() {
         countSelectedIndexChangeEvent = 0;
         countSelectedItemChangeEvent = 0;
         countSelectedIndicesChangeEvent = 0;
         countSelectedItemsChangeEvent = 0;
 
-        apple.getChildren().addAll(new TreeItem("Apple3"), new TreeItem("Apple2"), new TreeItem("Apple1"));
-        apple.setExpanded(true);
-        orange.getChildren().addAll(new TreeItem("Orange3"), new TreeItem("Orange2"), new TreeItem("Orange1"));
-        orange.setExpanded(true);
-        banana.getChildren().addAll(new TreeItem("Banana3"), new TreeItem("Banana2"), new TreeItem("Banana1"));
-        banana.setExpanded(true);
+        TreeTableColumn<String, String> col = new TreeTableColumn<String, String>("column");
+        col.setSortType(DESCENDING);
+        col.setCellValueFactory(param -> new ReadOnlyObjectWrapper<String>(param.getValue().getValue()));
+        treeTableView.getColumns().add(col);
+
+        TreeItem<String> treeRoot = new TreeItem<String>("root");
+        treeRoot.setExpanded(true);
+        treeTableView.setRoot(treeRoot);
+
+        final int FIRST_LEVEL_COUNT = 8;
+        for (int i = 0; i < FIRST_LEVEL_COUNT; i++) {
+            TreeItem<String> ti = new TreeItem<>( "" + i);
+            ti.setExpanded(true);
+            treeRoot.getChildren().add(ti);
+
+            for (int j = 0; j < FIRST_LEVEL_COUNT - 1; j++) {
+                TreeItem<String> tj = new TreeItem<>("" + i + j);
```

Comment:
The tree item strings will not be unique. This won't affect whether the test passes or fails, since `TreeItem` does not override `equals`, but it might be better if the strings were unique (in case there ever was an error it would be easier to understand).

Reply:
Hi Kevin, I verified the the tree item strings again. They seem to be unique. A total of 8800 of tree items get created each with unique string. Could you please recheck.

Reply:
Confirmed. This was my mistake.